lambdas/store-identity/src/main/java/uk/gov/di/ipv/core/storeidentity/StoreIdentityHandler.java

```diff
@@ -0,0 +1,101 @@
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_ERROR_PATH;
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_IDENTITY_STORED_PATH;
+
+public class StoreIdentityHandler implements RequestHandler<ProcessRequest, Map<String, Object>> {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Map<String, Object> JOURNEY_IDENTITY_STORED =
+            new JourneyResponse(JOURNEY_IDENTITY_STORED_PATH).toObjectMap();
+    private final ConfigService configService;
+    private final IpvSessionService ipvSessionService;
+    private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
+    private final SessionCredentialsService sessionCredentialsService;
+    private final VerifiableCredentialService verifiableCredentialService;
+
+    public StoreIdentityHandler(
+            ConfigService configService,
+            ClientOAuthSessionDetailsService clientOAuthSessionDetailsService,
+            IpvSessionService ipvSessionService,
+            SessionCredentialsService sessionCredentialsService,
+            VerifiableCredentialService verifiableCredentialService) {
+        this.configService = configService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.ipvSessionService = ipvSessionService;
+        this.sessionCredentialsService = sessionCredentialsService;
+        this.verifiableCredentialService = verifiableCredentialService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public StoreIdentityHandler() {
+        this.configService = new ConfigService();
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        this.sessionCredentialsService = new SessionCredentialsService(configService);
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public Map<String, Object> handleRequest(ProcessRequest input, Context context) {
+        LogHelper.attachComponentId(configService);
+        configService.setFeatureSet(RequestHelper.getFeatureSet(input));
+
+        try {
+            var ipvSessionItem =
+                    ipvSessionService.getIpvSession(RequestHelper.getIpvSessionId(input));
+            ClientOAuthSessionItem clientOAuthSessionItem =
+                    clientOAuthSessionDetailsService.getClientOAuthSession(
+                            ipvSessionItem.getClientOAuthSessionId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSessionItem.getGovukSigninJourneyId());
+
+            verifiableCredentialService.storeIdentity(
+                    sessionCredentialsService.getLatestCredentialsByIssuer(
+                            ipvSessionItem.getIpvSessionId(), clientOAuthSessionItem.getUserId()),
```

Comment:
This is probably fine for now, but when we introduce EVCS we may want to examine this logic a little bit - specifically whether 'latest VC from each issuer' is the right thing.

I'm wondering if instead we actually need a bit of logic that clears the current address VC from the session store if you select 'change address' - not quite sure where we'd put that logic though :/

Reply:
Maybe a new lambda that can get plugged into the journey map.

Reply:
Yeah, I can't quite decide between a new lambda vs. just a little bit of logic at the end of `ProcessCriCallback`... It probably depends how generic it is.

I can think of a few options I'd be happy with:
- Update to `BuildCriOauthRequest` that clears any existing VC for the CRI [if specified in the journey map?]
- Update to `ProcessCriCallback` that clears any existing VC for the CRI before storing the new one
- Looking ahead to https://govukverify.atlassian.net/browse/PYIC-5597, construct some more generic deletion lambda that can be instructed to specifically delete the old address (i.e. before going into the address CRI)

WDYT?

Reply:
Well I've implemented the ProcessCriCallback option here, but I think there's legs in the new lambda, especially if we wanted to make this more generic.
I don't think do it in the build-cri-oauth request lambda, as we need to make sure that we've cleared the dups for the build-user-identity lambda.

Reply:
Yeah, I think this is fine for now - and feels like the path of least resistance.

If we start needing a fancier deletion lambda (either for https://govukverify.atlassian.net/browse/PYIC-5597) or when handling COI journeys (that will also need to clear session storage once you start a name change), then we can update it.